### PR TITLE
Maxwell-Juttner Distribution

### PR DESCRIPTION
**Pull request by cameronjy**
This PR adds the Maxwell-Juttner distribution and the relevant parameter to WarpX. The plasma velocity is initialized in a frame moving with velocity beta with respect to the simulation frame using the Sobol method, but is transformed back to the simulation frame before being passed to the simulation using the flipping method, as described in Zenitani 2015. More information on parameters and references have been added in the parameters.rst file as well.

WarpX MJ and Analytic MJ Comparison

This is an example of the code producing a MJ distribution with a boost of beta = 0.3 in the y direction.

### CODE REVIEW

**Comment by RemiLehe** [comment]
Thanks a lot for this PR @cameronjy 
The automated tests indicate some style issues (presence of tabs in the code): see
https://github.com/ECP-WarpX/WarpX/pull/516/checks?check_run_id=295152395
Could you fix them and update the PR? Thanks!

**Comment by RevathiJambunathan** [comment]
Thanks for the PR, @cameronjy :)

File: Source/Initialization/InjectorMomentum.H [hunk 1]
```diff
@@ -47,6 +47,49 @@ private:
...
+            u[d] = -t*std::log(amrex::Random()*amrex::Random()*amrex::Random());
+            g = std::sqrt(1+std::pow(u[d],2));
+            x1 = t*std::log(amrex::Random());
+        }
+        x1 = amrex::Random();
+        x2 = amrex::Random();
+        u[(d+1)%3] = 2*u[d]*std::sqrt(x1*(1-x1))*std::sin(2*M_PI*x2);
+        u[(d+2)%3] = 2*u[d]*std::sqrt(x1*(1-x1))*std::cos(2*M_PI*x2);
+        u[d] = u[d]*(2*x1-1);
+        x1 = amrex::Random();
```
Comment (RevathiJambunathan) [inline Source/Initialization/InjectorMomentum.H]: @cameronjy Could you please add some comments here stating that this part does the volume-transformation and that we use the flipping method taking advantage of symmetry.

Comment (RemiLehe) [reply Source/Initialization/InjectorMomentum.H]: I agree with @RevathiJambunathan. Also: could you add the reference of the paper that describes this, in the comment.

Comment (cameronjy) [reply Source/Initialization/InjectorMomentum.H]: @RemiLehe @RevathiJambunathan I made the changes -- let me know if the comments make sense or if they are too lengthy. I can elaborate details of my comments with respect to the Zenitani paper in person as well.

**Comment by RemiLehe** [comment]
Thanks for this pull request!

**Comment by RevathiJambunathan** [comment]
Congrats on your first PR, @cameronjy

**Pull request merged**